### PR TITLE
Feature/create user class

### LINKED ISSUE #6: Add ability to favorite/unfavorite recipes
Allow a user to favorite or unfavorite recipes (add to / remove from the user’s favoriteRecipes)

### LINKED ISSUE #7: Add functionality to add recipes to recipesToCook area
Decide to cook a recipe that week (add to my recipesToCook)

### LINKED ISSUE #8: Add filtering of favoriteRecipes
- Filter my favoriteRecipes by a tag. (Extension option: filter by multiple tags)
- Filter my favoriteRecipes by its name. (Extension option: filter by name or ingredients)

### LINKED ISSUE #9: Add onLoad functionality
- As a user, I should be able to favorite / unfavorite recipes that I like and can easily find again.
- As a user, I should be able to filter my favorited recipes by a tag. (Extension option: filter by multiple tags)
- As a user, I should be able to search my favorited recipes by its name. (Extension option: search by name or ingredients)
- As a user, I should be able to add a recipe to a list of recipes to cook (this functionality will be necessary for Part Two of this project).

### PR DESCRIPTION
**Pull request by nhodnett**
- What's this PR do?
Allows a user to favorite or unfavorite recipes (add to / remove from the user’s favoriteRecipes)
Filters favoriteRecipes by a tag.
Filters favoriteRecipes by  name. 
Allows users to favorite / unfavorite recipes that they like and can easily find again.
Allows users to filter their favorited recipes by a tag.
Allows users to search their favorited recipes by its name.
Allows users to add a recipe to a list of recipes to cook.

- Where should the reviewer start?
scripts.js
index.html

- How should this be manually tested?
Run npm start when in the main directory
Go to http://localhost:8080/

- Any background context you want to provide?
N/A

- What are the relevant tickets?
resolve #6 
resolve #7 
resolve #8 
resolve #9 

- Screenshots (if appropriate)
N/A

- Questions:
None

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -47,11 +55,44 @@ <h1 class="home-btn">What's Cookin?</h1>
...
+        <option value="spread">spread</option>
+      </select>
+      <button class="filter-favorite-recipes-btn">Filter</button>
+    </section>
+
+    <section class="favorite-recipe-area">
+      <button class="unfavorite-btn hidden">Unfavorite</button>
     </section>
-    <section class="user-area hidden">
+
```
Comment (aominhlong) [inline dist/index.html:93]: Fix spacing

File: src/scripts.js [hunk 2]
```diff
@@ -18,35 +19,58 @@ const favoriteRecipeArea = document.querySelector(".favorite-recipe-area")
...
+recipesToCookButton.addEventListener('click', displayRecipesToCookArea)
+
 favoriteRecipeArea.addEventListener('click', (event) => {
   displayRecipe(event);
 })
+
+recipesToCookArea.addEventListener("click", (event) => {
+  displayRecipe(event);
+})
 filterButton.addEventListener('click', (event) => {
```
Comment (aominhlong) [inline src/scripts.js:64]: Fix spacing between event listeners

File: src/scripts.js [hunk 3]
```diff
@@ -18,35 +19,58 @@ const favoriteRecipeArea = document.querySelector(".favorite-recipe-area")
 const favoritesAreaButton = document.querySelector(".favorites-btn")
 const favoriteRecipeButton = document.querySelector(".favorite-btn")
 const displayRecipeSection = document.querySelector(".display-recipe")
-const userAreaButton = document.querySelector(".user-btn")
-const userArea = document.querySelector(".user-area")
+const recipesToCookButton = document.querySelector(".recipes-to-cook-btn")
```
Comment (mattruder) [inline src/scripts.js:22]: lets work on ordering this section when we refactor

File: src/scripts.js [hunk 4]
```diff
@@ -18,35 +19,58 @@ const favoriteRecipeArea = document.querySelector(".favorite-recipe-area")
...
 
 viewAllButton.addEventListener('click', viewAllRecipes)
 homeButton.addEventListener('click', goHome)
-// grabRecipe.addEventListener('click', displayRecipe)
 allRecipeList.addEventListener("click", (event) => {
   displayRecipe(event);
 })
 displayRecipeSection.addEventListener("click", (event) => {
   addRecipeToFavorites(event);
 })
```
Comment (mattruder) [inline src/scripts.js:48]: the event listener section can be refactored as well

**Pull request merged**